File: Inventory-app/android/app/src/main/java/com/wekanmdb/storeinventory/model/job/Jobs.kt
package com.wekanmdb.storeinventory.model.job

import com.wekanmdb.storeinventory.model.product.Products
import com.wekanmdb.storeinventory.model.store.Stores
import com.wekanmdb.storeinventory.model.user.Users
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass
import org.bson.types.ObjectId
import java.util.*

open class Jobs(
        @PrimaryKey
        var _id: ObjectId = ObjectId(),
        var createdBy: Users? = null,
        var assignedTo: Users? = null,
        var sourceStore: Stores? = null,
        var destinationStore: Stores? = null,
        var pickupDatetime: Date? = null,
        var status: String = "",
        var products: RealmList<ProductQuantity>? = null,
        var _partition: String = "master"
) : RealmObject()


open class ProductQuantity(
        @PrimaryKey
        var _id: ObjectId = ObjectId(),
        var product: Products ?= null,
        var quantity: Int = 0,
        var _partition: String = "master"
) : RealmObject()

File: Inventory-app/ios/AppShowcase/MVVM/Models/Jobs.swift
//
//  Jobs.swift
//  AppShowcase
//
//  Created by Gagandeep on 06/09/21.
//

import Foundation
import RealmSwift

class Jobs: Object {
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var _partition: String = ""
    @Persisted var status: String = ""
    @Persisted var createdBy: Users? = nil
    @Persisted var assignedTo: Users? = nil
    @Persisted var sourceStore: Stores? = nil
    @Persisted var destinationStore: Stores? = nil
    @Persisted var pickupDatetime: Date? = nil
    @Persisted var products: RealmSwift.List<ProductQuantity>

    convenience init(_partition: String, status: String, createdBy: Users?, assignedTo: Users?, sourceStore: Stores?, destinationStore: Stores?, pickupDatetime: Date?, products: List<ProductQuantity>) {
        self.init()
        self._partition = _partition
        self.status = status
        self.createdBy = createdBy
        self.assignedTo = assignedTo
        self.sourceStore = sourceStore
        self.destinationStore = destinationStore
        self.pickupDatetime = pickupDatetime
        self.products = products
    }
}
